filter-processor/src/test/java/org/springframework/cloud/stream/module/filter/FilterProcessorIntegrationTests.java

```diff
@@ -81,6 +81,56 @@ public void test() throws InterruptedException {
 			channels.input().send(new GenericMessage<Object>("hi!"));
 			assertThat(collector.forChannel(channels.output()), receivesPayloadThat(is("hello world")));
 			assertThat(collector.forChannel(channels.output()).poll(10, MILLISECONDS), is(nullValue(Message.class)));
+
+			channels.input().send(new GenericMessage<Object>("hi!"));
+			channels.input().send(new GenericMessage<Object>("hello world"));
+			assertThat(collector.forChannel(channels.output()), receivesPayloadThat(not("hi!")));
+			assertThat(collector.forChannel(channels.output()).poll(10, MILLISECONDS), is(nullValue(Message.class)));
+
+		}
+	}
+
+	@WebIntegrationTest("expression=payload.contains('foo')")
+	public static class UsingContainsExpressionIntegrationTests extends FilterProcessorIntegrationTests {
+
+		@Test
+		public void test() throws InterruptedException {
+
+			channels.input().send(new GenericMessage<Object>("bazbar"));
+			channels.input().send(new GenericMessage<Object>("barbaz"));
+			channels.input().send(new GenericMessage<Object>("foobar"));
+			assertThat(collector.forChannel(channels.output()), receivesPayloadThat(is("foobar")));
+			assertThat(collector.forChannel(channels.output()).poll(10, MILLISECONDS), is(nullValue(Message.class)));
+
+			channels.input().send(new GenericMessage<Object>("''babfoo''"));
+			assertThat(collector.forChannel(channels.output()), receivesPayloadThat(is("''babfoo''")));
+			assertThat(collector.forChannel(channels.output()).poll(10, MILLISECONDS), is(nullValue(Message.class)));
+
+			channels.input().send(new GenericMessage<Object>("bazbar"));
+			channels.input().send(new GenericMessage<Object>("foobar"));
+			assertThat(collector.forChannel(channels.output()), receivesPayloadThat(not("bazbar")));
+			assertThat(collector.forChannel(channels.output()).poll(10, MILLISECONDS), is(nullValue(Message.class)));
+		}
+	}
+
+	@WebIntegrationTest("expression=payload=='baz'")
```

Comment:
I wonder why do we need these tests?
Looks like they check the SpEL stuff, not the `filter` processor...

There is really enough just one `expression` test on the matter.